tests/python/ring_test.py

```
@@ -0,0 +1,104 @@
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from absl.testing import absltest
+
+from prime_ir.mlir.ir import *
```

Ruff (0.16.1)

[error] `from prime_ir.mlir.ir import *` used; unable to detect undefined names

(F403)